hack/generator/pkg/astmodel/armconversion/convert_to_arm_function_builder.go

```diff
@@ -184,6 +185,134 @@ func (builder *convertToARMBuilder) referencePropertyHandler(
 	)
 }
 
+func (builder *convertToARMBuilder) flattenedPropertyHandler(
+	toProp *astmodel.PropertyDefinition,
+	fromType *astmodel.ObjectType) []dst.Stmt {
+
+	toPropName := toProp.PropertyName()
+
+	// collect any fromProps that were flattened from the to-prop
+	var fromProps []*astmodel.PropertyDefinition
+	for _, prop := range fromType.Properties() {
+		if prop.WasFlattenedFrom(toPropName) {
+			fromProps = append(fromProps, prop)
+		}
+	}
+
+	// there are none to copy; exit
+	if len(fromProps) == 0 {
+		return nil
+	}
+
+	allTypes := builder.codeGenerationContext.GetAllReachableTypes()
+
+	// the toProp shape here must be:
+	// 1. maybe a typename, pointing to…
+	// 2. maybe optional, wrapping …
+	// 3. maybe a typename, pointing to…
+	// 4. an object type
+
+	// (1.) resolve the outer typename
+	toPropType, err := allTypes.FullyResolve(toProp.PropertyType())
+	if err != nil {
+		panic(err)
+	}
+
+	needToInitializeToProp := false // we need to init the target if it is optional
+	var toPropTypeName astmodel.TypeName
+	// (2.)  resolve any optional type
+	if optType, ok := toPropType.(*astmodel.OptionalType); ok {
+		needToInitializeToProp = true
+		// (3.) resolve any inner typename
+		toPropTypeName = optType.Element().(astmodel.TypeName)
+		toPropType, err = allTypes.FullyResolve(optType.Element())
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	// (4.) we have found the underlying object type
+	toPropObjType := toPropType.(*astmodel.ObjectType)
+
+	// *** Now generate the code! ***
+
+	// Build the initializer for the to-prop (if needed)
+	var result []dst.Stmt
+	if needToInitializeToProp {
+		result = []dst.Stmt{builder.buildToPropInitializer(fromProps, toPropTypeName, toPropName)}
+	}
+
+	// Copy each from-prop into the to-prop
+	for _, fromProp := range fromProps {
+		// find the corresponding inner property on the to-prop type
+		toSubProp, ok := toPropObjType.Property(fromProp.PropertyName())
+		if !ok {
+			panic("unable to find source of flattened property")
+		}
+
+		// generate conversion
+		stmts := builder.typeConversionBuilder.BuildConversion(astmodel.ConversionParameters{
+			Source:            astbuilder.Selector(dst.NewIdent(builder.receiverIdent), string(fromProp.PropertyName())),
+			SourceType:        fromProp.PropertyType(),
+			Destination:       astbuilder.Selector(astbuilder.Selector(dst.NewIdent(builder.resultIdent), string(toPropName)), string(toSubProp.PropertyName())),
+			DestinationType:   toSubProp.PropertyType(),
+			NameHint:          string(toSubProp.PropertyName()),
+			ConversionContext: nil,
+			AssignmentHandler: nil,
+			Locals:            builder.locals,
+		})
+
+		// failed?
```

Comment:
Same q as other file - this comment makes it not clear if this is expected or an error case. Seems expected given you're not panicing? Might be worth noting when this can happen.